lab-14-chris/test/server-test.js

```diff
@@ -0,0 +1,286 @@
+          expect(res.body._id).to.equal(songObj[0]._id);
+
+          done();
+        });
+      });
+
+      it('should repsond with a 404 no id entered', done => {
+        chai.request(server)
+        .get(`/api/song id=null`)
+        .end((err, res) => {
+          console.log(res.status);
+          expect(res.status).to.equal(404);
+          done();
+        });
+      });
+
+      // it('should return the song with proper title property', done => {
+      //
+      //     expect(res.body.title).to.equal('test-title');
+      //   });
+      //
+      //   done();
+      // });
+    });
+  });
+
+  describe('PUT method', function() {
+
+    describe('/api/song endpoint', function() {
+
+  //     it('should respond with a 200 on proper request', done => {
+  //       chai.request(server)
+  //       .put('/api/song?id=test-post&title=newtest-title&artist=newtest-artist&album=newtest-album')
+  //       .end((err, res) => {
+  //
+  //         expect(res.status).to.equal(200);
+  //         done();
+  //       });
+  //     });
+  //
+  //     it('should "put" changes to song property title', done => {
+  //       fs.readFile('./data/json-storage/test-post.json', (err, data) => {
+  //         data = JSON.parse(data.toString());
+  //
+  //         expect(data.title).to.equal('newtest-title');
+  //       });
+  //
+  //       done();
+  //     });
+  //
+  //     it('should "put" changes to song property artist', done => {
+  //       fs.readFile('./data/json-storage/test-post.json', (err, data) => {
+  //         data = JSON.parse(data.toString());
+  //
+  //         expect(data.artist).to.equal('newtest-artist');
+  //       });
+  //
+  //       done();
+  //     });
+  //
+  //     it('should "put" changes to song property album', done => {
+  //       fs.readFile('./data/json-storage/test-post.json', (err, data) => {
+  //         data = JSON.parse(data.toString());
+  //
+  //         expect(data.album).to.equal('newtest-album');
+  //       });
+  //
+  //       done();
+  //     });
+  //
+  //     it('should save to json file of same id name with changes', done => {
+  //
+  //       expect('../data/json-storage/test-post.json').to.exist;
+  //
+  //       done();
+  //     });
+  //   });
+  // });
+  //
+  // describe('DELETE method', function() {
+  //
+  //   describe('/api/song endpoint', function() {
+  //
+  //     it('should respond with a 200 on proper request', done => {
+  //       chai.request(server)
+  //       .delete('/api/song?id=test-post')
+  //       .end((err, res) => {
+  //         expect(res.status).to.equal(200);
+  //       });
+  //       done();
+  //     });
+  //
+  //     it('should delete the file by id selected', done => {
+  //       fs.readFile('./data/json-storage/test-post.json', (err, data) => {
+  //         expect(data).to.be('undefined');
+  //         expect(err);
+  //       });
+  //       done();
+  //     });
+    });
+  });
+
+  after(done => {
+    app.close();
```

Comment:
how could you use your after block more effectively? stretch: clean up your database after EVERY test.